eth-providers/src/base-provider.ts

```diff
@@ -766,15 +789,45 @@ export abstract class BaseProvider extends AbstractProvider {
 
   // Queries
   getTransaction = (transactionHash: string): Promise<TransactionResponse> => throwNotImplemented('getTransaction');
+
+  getTransactionByHash = async (transactionHash: string): Promise<TX> => {
+    const tx = await getTxReceiptByHash(transactionHash);
+
+    if (!tx) {
+      return logger.throwError(`transaction hash not found`, Logger.errors.UNKNOWN_ERROR, { transactionHash });
+    }
+
+    const nonce = await this.getEvmTransactionCount(tx.from, tx.blockHash);
+    const extrinsic = await this._getExtrinsicByHashAtBlock(transactionHash, tx.blockHash);
+
+    const { args } = JSON.parse(extrinsic! as any).method;
```

Comment:
better to check not undefined and return a real error or throw assertion error indicating it should never happen.
why `JSON.parse` is needed? `extrinsic` should be `GenericExtrinsic`?

Reply:
without JSON.pase, extrinsic looks something like 
```
Type {
  registry: TypeRegistry { createdAtHash: undefined },
  createdAtHash: undefined,
  _raw: Type(2) [Map] {
    'signature' => Type(5) [Map] {
      'signer' => [Type],
      'signature' => [Type],
      'era' => [Type],
      'nonce' => [Compact],
      'tip' => [Compact],
      registry: [TypeRegistry],
      createdAtHash: undefined
    },
    'method' => Type(2) [Map] {
      'callIndex' => [GenericCallIndex [Uint8Array]],
      'args' => [Struct [Map]],
      registry: [TypeRegistry],
      createdAtHash: undefined,
      _meta: [Type [Map]]
    },
    registry: TypeRegistry { createdAtHash: undefined },
    createdAtHash: undefined
  }
}
```
and after JSON.parse it looks like 
```
{
  signature: {
    signer: { id: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY' },
    signature: {
      sr25519: '0xae8968c6bfd626d0745c4bc82ca427f47aefe710b58aa23a2bdaa32607af406096388b12d4548badff48fc18e2f89c5de45361a794c99cd15684b46567236c8e'
    },
    era: { mortalEra: '0x3502' },
    nonce: 5,
    tip: 0
  },
  method: {
    callIndex: '0xb401',
    args: {
      target: '0x0230135fded668a3f7894966b14f42e65da322e4',
      input: '0x3d8d962000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000e8d4a510000000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000010000010000000000000000000000000000000000000000000000000000000001000000',
      value: 1000000000000,
      gas_limit: 2000001,
      storage_limit: 1
    }
  }
}
```

on previous line we checked `if (!tx) { throw error }`, so I was assuming now tx exist, but let's be more safe the check extrinsic exist, I will add the check

Reply:
- added check to make sure extrinsic exist
- to native `toJSON` instead of `JSON.parse()`